lib/ecto/schema.ex

```diff
@@ -841,7 +843,8 @@ defmodule Ecto.Schema do
       `:delete_all` will only remove data from the join source, never the
       associated records. Notice `:on_delete` may also be set in migrations
       when creating a reference. If supported, relying on the database via
-      migrations is preferred
+      migrations is preferred. `:nilify_all` and `:delete_all` will not cascade to child
+      records unless set via database migrations.
```

Comment:
Is it possible someone will bump into something similar that happened to me on this type of association? For example, where the join source is another Ecto model that has a third column referencing a child model that has `on_delete: :delete_all` set?

I guess my confusion was in assuming the `on_delete` callbacks would cascade down the association tree. My interpretation of the docs above is that it won't _automatically_ remove data from the target associations, but since it doesn't mention what happens if those associations have `on_delete` callbacks on them, I'd probably still experience the same confusion. So to me, there is a difference between what was there and what I added.

Anyway, up to you--I can remove this added note if you strongly disagree :)